gemrb/GUIScripts/InventoryCommon.py

```diff
@@ -363,32 +395,48 @@ def DisplayItem (slotItem, type):
 
 	# in pst one can cycle through all the items from the description window
 	if GameCheck.IsPST():
-		def DisplayNext(dir):
-			import GUIINV
-
-			id = GemRB.GetVar('ItemButton') + dir;
-			while id not in GUIINV.ItemHash or not GUIINV.ItemHash[id][1]:
-				id += dir;
-				if id < 0:
-					id = 43
-				elif id > 43:
-					id = 0
-
-			GemRB.SetVar('ItemButton', id)
-			CloseItemInfoWindow()
-			OpenItemInfoWindow(None, GUIINV.ItemHash[id][0])
 
 		#left scroll
 		Button = Window.GetControl (13)
-		Button.SetEvent (IE_GUI_BUTTON_ON_PRESS, lambda: DisplayNext(-1))
+		Button.SetEvent (IE_GUI_BUTTON_ON_PRESS, lambda: CycleDisplayItem(-1))
 
 		#right scroll
 		Button = Window.GetControl (14)
-		Button.SetEvent (IE_GUI_BUTTON_ON_PRESS, lambda: DisplayNext(1))
+		Button.SetEvent (IE_GUI_BUTTON_ON_PRESS, lambda: CycleDisplayItem(1))
 
 	ItemInfoWindow.ShowModal(MODAL_SHADOW_GRAY)
 	return
 
+def CycleDisplayItem(dir):
+
+	ib = int(GemRB.GetVar('ItemButton'))
+	thisitem = ib
+
+	#print "nextid: ", ib
+
+	pc = GemRB.GameGetSelectedPCSingle ()
+
+	slot_item = None
+	
+	#try the next slot for an item. if the slot is empty, loop until one is found.
+	while not slot_item:
+		ib+=dir
```

Comment:
```suggestion
		ib += dir
```

Reply:
I'll squash this one in when I update it